tests/conditional_compilation/conftest.py

```diff
@@ -66,20 +67,26 @@ def pytest_generate_tests(metafunc):
     params = []
     ids = []
 
-    with open(metafunc.config.getoption('test_conf'), "r") as file:
+    test_config = metafunc.config.getoption('test_conf')
+    with open(test_config, "r") as file:
         test_cases = yaml.safe_load(file)
 
     for test in test_cases:
         extra_args = {}
         model_path = test["model"]["path"]
         if "marks" in test:
             extra_args["marks"] = test["marks"]
+        
+        path_with_env_vars = Path(expand_env_vars(model_path))
+        model_path.replace('$', '').replace('{', '').replace('}', '')
+        params.append(pytest.param(model_path, path_with_env_vars, **extra_args))
+        ids = ids + [model_path]
 
-        test_id = model_path.replace('$', '').replace('{', '').replace('}', '')
-        params.append(pytest.param(test_id, Path(expand_env_vars(model_path)), **extra_args))
-        ids = ids + [test_id]
     metafunc.parametrize("test_id, model", params, ids=ids)
 
+    with open(Path(__file__).parent / "cc_tests.json", "w") as file:
```

Comment:
current implementation still dump all the cases which defined in config. But I guess we want to dump only cases which are exactly run. 

Also it would be great to track status of execution - in case of failures there is no need to continue test execution (means build of minimized build and run test_infer). It may be also done by checking .csv files exist, so need to find the best solution